### PR TITLE
record search: implement ng-content on page

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds ng-content on result page.
* Adds a new event to transmit the result of search.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Extracted translations?

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts [hunk 1]
```diff
@@ -215,12 +216,20 @@ export class RecordSearchComponent implements OnInit, OnDestroy {
...
   }
 
+  /**
+   * Result of search with type and records (Elasticsearch)
+   * @param event, EsResult content type and records
+   */
+  recordsResult(event: EsResult) {
+    // Implement this function on your extended class.
+  }
+
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts:226]: I think you should not add this method in ng-core, only in extended class.

File: projects/rero/ng-core/src/lib/record/search/record-search-page.component.html [hunk 2]
```diff
@@ -15,6 +15,6 @@
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 -->
 <ng-core-record-search [adminMode]="adminMode" [currentType]="currentType" [types]="types" [detailUrl]="detailUrl"
-  [showSearchInput]="showSearchInput" [q]="q" [page]="page" [size]="size"
-  [sort]="sort" (parametersChanged)="updateUrl($event)">
+  [showSearchInput]="showSearchInput" [q]="q" [page]="page" [size]="size" [sort]="sort"
+  (parametersChanged)="updateUrl($event)" (esResult)="recordsResult($event)">
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search-page.component.html:19]: esResult should not be configured in ng-core, only in the extended component.

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts [hunk 3]
```diff
@@ -172,6 +172,11 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
    */
   @Output() parametersChanged = new EventEmitter<any>();
 
+  /**
+   * Output current result of records by type
+   */
+  @Output() esResult = new EventEmitter<EsResult>();
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/search/record-search.component.ts:178]: A better name would be `recordSearched` or something like this, it must represent an event/action.

File: projects/rero/ng-core/src/lib/record/record.ts [hunk 4]
```diff
@@ -24,6 +24,11 @@ export class Record {
   links: any;
 }
 
+export class EsResult {
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/lib/record/record.ts:27]: I would prefer `SearchResult`, but no matter.

File: projects/rero/ng-core/src/public-api.ts [hunk 5]
```diff
@@ -67,4 +69,4 @@ export * from './lib/utils/sort-by-keys';
 export * from './lib/utils/utils';
 export * from './lib/validator/time.validator';
 export * from './lib/validator/unique.validator';
-export * from './lib/error/error.component';
```
Comment (sebastiendeleze) [inline projects/rero/ng-core/src/public-api.ts:70]: Why removing this export, I think it is needed.

**Review by sebastiendeleze — CHANGES_REQUESTED** [review]

File: projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts (hunk 1 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/search/record-search-page.component.ts:226]: done

File: projects/rero/ng-core/src/lib/record/search/record-search-page.component.html (hunk 2 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/search/record-search-page.component.html:19]: done

File: projects/rero/ng-core/src/lib/record/search/record-search.component.ts (hunk 3 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/search/record-search.component.ts:178]: done

File: projects/rero/ng-core/src/public-api.ts (hunk 5 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/public-api.ts:70]: done

File: projects/rero/ng-core/src/lib/record/record.ts (hunk 4 above)
Comment (Garfield-fr) [reply projects/rero/ng-core/src/lib/record/record.ts:27]: done

**Review by zannkukai — APPROVED** [review]

**Review by sebastiendeleze — APPROVED** [review]

**Review by benerken — APPROVED** [review]

**Review by lauren-d — APPROVED** [review]

**Pull request merged**